terraform/ec2/amis.tf

```diff
@@ -97,6 +99,9 @@ variable "amis" {
       os_family = "debian"
       ami_search_pattern = "debian-10-amd64*"
       ami_owner = "679593333241"
+      # NOTE: we need product code to pick the right debian 10.
+      ami_product_code = [
+        "auhljmclkudu651zy27rih2x2"]
```

Comment:
Q: why do we only need to apply `product_code` filter for this AMI? How about the others?

Reply:
For AMI owned by amazon, there is no product code, and if you apply the filter with `*`, the API returns nothing. I didn't check other marketplace AMI, based on subscription in the test account we only have subscription for Debian 10 and CentOS.

The reason we need product code for debian 10 is the filter pattern alone will return two products, one from Debian, one from dshop. Those two types of AMIs can only be distinguished using product code.